schema-engine/connectors/schema-connector/src/schema_connector.rs

```diff
@@ -75,6 +85,10 @@ pub trait SchemaConnector: Send + Sync + 'static {
     /// Return the schema dialect of the connector.
     fn schema_dialect(&self) -> Box<dyn SchemaDialect>;
 
+    /// The default namespaces for the connector if it supports multiple namespaces.
+    /// Should be derived from the connectors runtime configuration but can fallback to the dialect's default.
+    fn default_namespace(&self) -> Option<&str>;
```

Comment:
I'd call this something else to avoid confusion
```suggestion
    fn default_runtime_namespace(&self) -> Option<&str>;
```